### ISSUE TITLE
Dagger Training Stableness

### ISSUE THREAD

**Issue by HuskyKingdom** [jacobkrantz/VLN-CE#64]
Hi Jacob,

Thanks for your interesting work!
I was recently trying my model on VLNCE, specifically, I have been following the IL+Dagger training schedule like the original paper. It is found that using pure dagger training on the original data (without finetuning and PM) to train the model from iteration 0-9, the loss curve initially decreases during the first few iterations when P is large (hence more chance of selecting GT actions), but it shows an increasing trend in later iterations (iteration 4-5 as shown below, further iterations are still under training).

It is somehow making sense given the fundamental idea behind the Dagger training, but I just wanna ask if you had anything similar in your experiments, or could it be the problem on my side? 

Thanks for your time and looking forward to your reply. Please also let me know if you need further information.

![image](https://github.com/user-attachments/assets/1722de20-6c7b-4232-8330-f3368f6d3619)
![image](https://github.com/user-attachments/assets/c7f8ecfa-3062-4da8-b671-1bb11cda12c1)
![image](https://github.com/user-attachments/assets/021d1138-ef57-4e4c-b730-d2307760f2f7)
![image](https://github.com/user-attachments/assets/24acbf67-ac23-41c4-8430-4d2b2723b860)
![image](https://github.com/user-attachments/assets/c4b19b95-b4bc-4dcc-b531-a089f18e5cf8)

**Comment by BaiLing09** [comment]
Hello, can you download the RxR_VLNCE_v0.zip? I can't open this link. If you can download, could you send this RxR episodes dataset to me by the e-mail? My e-mail is 3309321488@qq.com. Thanks!
![屏幕截图 2025-01-02 170200](https://github.com/user-attachments/assets/070eb2a9-3e71-4829-bc53-631df69eebcc)

**Comment by bryce-wan** [comment]
> Hello, can you download the RxR_VLNCE_v0.zip? I can't open this link. If you can download, could you send this RxR episodes dataset to me by the e-mail? My e-mail is [3309321488@qq.com](mailto:3309321488@qq.com). Thanks! ![屏幕截图 2025-01-02 170200](https://private-user-images.githubusercontent.com/88820884/399658298-070eb2a9-3e71-4829-bc53-631df69eebcc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMzU0MTUsIm5iZiI6MTczNjIzNTExNSwicGF0aCI6Ii84ODgyMDg4NC8zOTk2NTgyOTgtMDcwZWIyYTktM2U3MS00ODI5LWJjNTMtNjMxZGY2OWVlYmNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDA3MzE1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4OThkODYwMDc5YWNiNGRjMWMwMjdlNmZiZTk3NzIyNmI2NDFlMWJhYWJmZDU5ZjBkOTQyYzQ2NzRlZGFhZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.C0pgwXmfD4NWEukg410qN_3qjjitEza9Q-jeYHqLngc)

The link is expired and I have sent you a copy I downloaded before. Please check your mailbox : )

**Comment by jacobkrantz** [comment]
Hey @HuskyKingdom, 
My intuition from observing your loss curves is that as the GT action probability decays, the agent encounters increased novel states in the state space. If the policy gathers more data before learning useful behavior from the previous dagger iteration, the resulting dataset will contain nonsensical training samples (e.g. "stuck" against walls). These samples may lead your policy to converge on undesirable behaviors. Ideas to avoid this include changing the decay function or increasing training epochs per DAgger iteration. Evaluate these models on a validation split though, because DAgger loss does not necessarily correlate with actual navigation performance. Good luck!

**Comment by jacobkrantz** [comment]
The RxR data link has been fixed (729d141b2ee10628061ada74dd3a5b9f70faeba5). Thanks for drawing attention to this!
